src/Kdyby/Doctrine/QueryBuilder.php

```diff
@@ -236,10 +236,9 @@ private function autoJoin(&$key)
 
 			$aliasLength = 1;
 			do {
-				$joinAs = substr($property, 0, $aliasLength++);
+				$joinAs = $this->buildJoinAs($property, $aliasLength);
```

Comment:
I think it would be better to always take first letter and suffix it with incremental number utill it finds a free alias